pages/_app.tsx

```diff
@@ -29,6 +29,7 @@ function Content({ children }: { children: ReactNode }) {
   const { session, isInstructor, error } = useSessionInit();
   const trigger = useScrollTrigger();
 
+  if (!isInstructor && router.pathname === "/book") return <>{children}</>;
```

Comment:
この場合のchildrenってFragmentでラップしなくてもいけますかね...？

```suggestion
  if (!isInstructor && router.pathname === "/book") return children;
```

Reply:
プリミティブ値への対処が必要なので Fragment をつかっていましたが children の型を ReactElement 等に変更すればいけます。

Reply:
> プリミティブ値への対処が必要なので Fragment をつかっていました

たしかにそれならFragmentを使用する必要性がありそうですね。

>  children の型注釈を ReactElement から変更すればいけます。

今より明確によりよいというわけではなさそうかと思いました。ありがとうございます。

Reply:
ReactElement ではなく、ReactNode でした。訂正します。
そして ReactNode より ReactChild などより厳格な型注釈を使うほうがよいかもですね。
参考文献: https://fettblog.eu/react-types-for-children-are-broken/

Reply:
確かに ReactChild でより厳密な型注釈になり ReactText が含まれうることが意図として表明できていそうですね :+1:

私は私でこの記事を眺めていました https://dackdive.hateblo.jp/entry/2019/08/07/090000